app/src/main/java/com/launcher/silverfish/launcher/appdrawer/AppArrayAdapter.java

```diff
@@ -113,8 +112,7 @@ public void onClick(View view) {
                         e.printStackTrace();
                     }
                 } else {
-                    i.setAction(Intent.ACTION_MAIN);
-                    i.setComponent(new ComponentName(app.packageName.toString(), app.activityName.toString()));
+                    i = mPackageManager.getLaunchIntentForPackage(app.packageName.toString());
```

Comment:
Ah, so that's the way it's done. Cool, thanks.

Reply:
You're welcome :)